cmake/modules/FindZephyr-sdk.cmake

```diff
@@ -60,14 +60,50 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
       set(ZEPHYR_TOOLCHAIN_VARIANT ${ZEPHYR_CURRENT_TOOLCHAIN_VARIANT})
     endif()
   else()
-    find_package(Zephyr-sdk ${Zephyr-sdk_FIND_VERSION} REQUIRED QUIET CONFIG PATHS
+    # Search for Zephyr SDK version 0.0.0 which does not exist, this is needed to
+    # return a list of compatible versions and find the best suited version that
+    # is available.
+    find_package(Zephyr-sdk 0.0.0 EXACT QUIET CONFIG PATHS
                  /usr
                  /usr/local
                  /opt
                  $ENV{HOME}
                  $ENV{HOME}/.local
                  $ENV{HOME}/.local/opt
                  $ENV{HOME}/bin)
+
+    # Remove duplicat entries and sort naturally in descending order.
+    list(REMOVE_DUPLICATES Zephyr-sdk_CONSIDERED_VERSIONS)
+    list(SORT Zephyr-sdk_CONSIDERED_VERSIONS COMPARE NATURAL ORDER DESCENDING)
+
+    # Loop over each found Zepher SDK version until one is found that is compatible.
+    foreach(ZEPHYR_SDK_CANDIDATE ${Zephyr-sdk_CONSIDERED_VERSIONS})
+      find_package(Zephyr-sdk ${ZEPHYR_SDK_CANDIDATE} EXACT QUIET CONFIG PATHS
+                   /usr
+                   /usr/local
+                   /opt
+                   $ENV{HOME}
+                   $ENV{HOME}/.local
+                   $ENV{HOME}/.local/opt
+                   $ENV{HOME}/bin)
+
+      if ("${Zephyr-sdk_FOUND}" EQUAL "1")
+        break()
+      endif()
+    endforeach()
+
+    if (NOT ("${Zephyr-sdk_FOUND}" EQUAL "1"))
+      # This means no compatible Zephyr SDK versions were found, set the version
+      # back to the minimum version so that it is displayed in the error text.
+      find_package(Zephyr-sdk ${Zephyr-sdk_FIND_VERSION} EXACT REQUIRED QUIET CONFIG PATHS
+                   /usr
+                   /usr/local
+                   /opt
+                   $ENV{HOME}
+                   $ENV{HOME}/.local
+                   $ENV{HOME}/.local/opt
+                   $ENV{HOME}/bin)
```

Comment:
These default search paths are repeated three times in this file. It would be nice if this could be put into a variable (if possible).

Reply:
Fixed